src/extension.ts

```diff
@@ -1,318 +1,388 @@
+       * calls to GitHub. It provides convenient typings that can be helpful for using the API.
+       *
+       * Documentation on GitHub's REST API can be found here: https://docs.github.com/en/rest
+       */
+      const octokit = await credentials.getOctokit();
+      const userInfo = await octokit.users.getAuthenticated();
+
+      vscode.window.showInformationMessage(
+        `Logged into GitHub as ${userInfo.data.login}`
+      );
+    }
+  );
+  context.subscriptions.push(disposable);
+  context.subscriptions.push(
+    vscode.commands.registerCommand("watermelon.start", async () => {
+      let config = await (
+        await gitAPI?.repositories[0]?.getConfig("remote.origin.url")
+      )?.split("/");
+      console.log(config);
+      if (config) {
+        repo = config[4].split(".")[0];
+        owner = config[3];
+      }
+      localUser = await gitAPI?.repositories[0]?.getGlobalConfig("user.name");
+
+      getPRsPerSHAs();
+      watermelonPanel.createOrShow(context.extensionUri);
+    })
+  );
+  vscode.window.onDidChangeTextEditorSelection((selection) => {
+    startLine = selection.selections[0].start.line;
+    endLine = selection.selections[0].end.line;
+    getSHAs();
+  });
+
+  if (vscode.window.registerWebviewPanelSerializer) {
+    // Make sure we register a serializer in activation event
+    vscode.window.registerWebviewPanelSerializer(watermelonPanel.viewType, {
+      async deserializeWebviewPanel(
+        webviewPanel: vscode.WebviewPanel,
+        state: any
+      ) {
+        // Reset the webview options so we use latest uri for `localResourceRoots`.
+        webviewPanel.webview.options = getWebviewOptions(context.extensionUri);
+        watermelonPanel.revive(webviewPanel, context.extensionUri);
+      },
+    });
+  }
 }
-function escapeFilePath(path:string| undefined){
-	if(path){
-		// $& means the whole matched string
-		return path.replace(/[. *\s+\ ?^${}()|[\]\\]/g, '\\$&');
-	}
-	else {return "";}
+function escapeFilePath(path: string | undefined) {
+  if (path) {
+    // $& means the whole matched string
+    return path.replace(/[. *\s+\ ?^${}()|[\]\\]/g, "\\$&");
+  } else {
+    return "";
+  }
 }
-
-async function getPRsPerSHAs(){
-
-	let owner = "";
-	// get repo name and owner basename
-	await exec(`cd ${escapeFilePath(folderRoute)} \n git config --get remote.origin.url`,
-		async function (error:string, stdout:string, stderr:string) {
-			const splitStdout = stdout.split("/");
-			// Retrieving the owner here is very important because it evades pulling issues from other repos
-			let localowner = splitStdout[3];
-			owner = localowner;
-			
-			// Get issues by commit hash and owner
-			await octokit.request(`GET /search/issues?type=Commits`, {
-				org: owner,
-				q: `hash:${arrayOfSHAs[0]}`
-			}).then(octoresp => {
-				const issuesBySHAs = octoresp.data.items;
-
-				// Show toast if no search results are found
```

Comment:
Don't delete this. Please re-add it so we keep showing this toast for better UX.

Reply:
Readded